Add management and delivery of sponsors of lunes

ulliholtgrave opened the pull request:
### Short description
<!-- Describe this PR in one or two sentences. -->
This PR adds the sponsors model that is delivered via the api and displayed within the app

### Proposed changes
<!-- Describe this PR in more detail. -->

- Add sponsor model
- Add sponsors to UI and Serializer
- Adjust REST-API to deliver sponsors with icons

### Resolved issues
<!-- List all issues which should be closed when this PR is merged. -->

Fixes: #371

JoeyStk commented on lunes_cms/cms/models/sponsor.py:
```diff
@@ -0,0 +1,18 @@
+    Model to manage our sponsors that are shown in the app.
+    """
+
+    name = models.TextField(
```
I think I find a TextField a bit to large. Maybe I'm not thinking of something here, but I think a CharField should be enough

JoeyStk commented on lunes_cms/cms/models/static.py:
```diff
@@ -66,6 +66,21 @@ def convert_umlaute_audio(instance, filename):
+    :return: file path for sponsor icons
+    :rtype: str
+    """
+    return create_ressource_path("sponsors", filename)
```
In a chat with @juliawohlhueter we thought it was better not to return the pathname but instead return a boolean and according icon (s. screenshot), if there is a logo or not. The same mechanism we have for vocabularies (incl. icon). 
![grafik](https://user-images.githubusercontent.com/72705147/219876771-ed356fdb-fff9-47df-bc87-0b28476d8067.png)

JoeyStk reviewed (changes requested):
Welcome to the Lunes CMS!
Thank you a lot for this PR and especially your immediate take to it.
I only have two minor suggestions :)

JoeyStk reviewed (changes requested):
Ah, and of course please add an entry to the changelog :)

ulliholtgrave commented:
@JoeyStk I addressed your mentioned issues and also adjusted some typos and general formatting issues (sorry, but I couldn't ignore my IDE warnings :see_no_evil: )

However, I couldn't really find out how you realize the display of the green checks for boolean fields. As I cannot really spent more time on that, I would like to ask you if you could address this representation issue, if that's okay for you :)

JoeyStk commented:
Yes, of course I can. I added my changes and I think it works as expected now. It would be great, if someone else could maybe check it out

timoludwig commented on lunes_cms/cms/models/sponsor.py:
```diff
@@ -0,0 +1,57 @@
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from .static import upload_sponsor_icons
+from ..utils import get_image_tag
```
```suggestion
from .static import upload_sponsor_logos
```

timoludwig reviewed (changes requested):
Thanks! :tada:
Also for all the spelling fixes :grin: 

I wondered whether the foreign key link from disciplines to sponsors was also part of the issue? The sponsors alone are not usable if they cannot be assigned to a discipline. We should probably either include this in this PR or open another issue.

timoludwig commented:
@JoeyStk after you changed your git history locally, e.g. by squashing commits, you have to force-push to the branch, otherwise the changes you made are partially reverted (like in this case).

timoludwig commented on lunes_cms/cms/models/discipline.py:
```diff
@@ -83,12 +85,13 @@ def image_tag(self):
         :rtype: str
         """
-        return self.title
+        return str(self.title)
```
This was reverted again

JoeyStk commented:
First of all, thank you very much for your feedback. Yes, I had screwed up something up while rebasing. But I hope it's all resolved now.

I updated the PR by doing the following things:
- Squash the commits
- Rebase from develop to have the most recent files, esp. changelog and migration files
- Undo the errors that I created while rebasing the first time

I hope that everything works now as expected :) 

> I wondered whether the foreign key link from disciplines to sponsors was also part of the issue? The sponsors alone are not usable if they cannot be assigned to a discipline. We should probably either include this in this PR or open another issue.

Good point, but as far as I understood the briefing the sponsor as not going to be linked to disciplines anymore, because this might cause legal troubles if Lunes is used in schools. I changed it in the issue now as well

timoludwig reviewed (approved):
Amazing, thanks! :pray:
